### PR TITLE
add wait_for loop for restore procedure

### PR DESCRIPTION
**Pull request by jdsn**
This should fix the errors we see with the backup and restore tests, because the restore is now happening in background and the current code did not wait for the restore to finish.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1547,9 +1547,20 @@ function jsonice()
     (echo -n '{}'; cat -) | sed -e 's/^{}\s*{/{/' | safely python -mjson.tool
 }
 
+function crowbar_any_status()
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: MINOR: I would rather call it `crowbar_api_status` but I think it is a matter of taste

**Comment by MaximilianMeister** [comment]
other than that +1

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4018,8 +4029,9 @@ function onadmin_crowbarrestore()
     if iscloudver 6plus ; then
         systemctl start crowbar.service
         wait_for 20 10 "onadmin_is_crowbar_api_available" "crowbar service to start"
-        crowbarctl backup upload /tmp/$btarball
-        crowbarctl backup restore $btarballname --yes
+        safely crowbarctl backup upload /tmp/$btarball --anonymous
+        safely crowbarctl backup restore $btarballname --anonymous --yes
+        wait_for 360 10 "crowbar_restore_status | grep -q '\"success\": *true'" "crowbar to be restored"
```
Comment (MaximilianMeister) [inline scripts/qa_crowbarsetup.sh]: MINOR: s/crowbar/Crowbar/

**Comment by rsalevsky** [comment]
+1

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: we agreed to change it later if it turns out to be confusing

**Comment by bmwiedemann** [comment]
+1 if the last 2 commits get some more explaination in the commit messages

**Pull request merged**